install.sh

```diff
@@ -192,11 +192,102 @@ legacy_transformations_end() {
 ############################################################
 legacy_transformations_repo_end() {
 
+    # Legacy values
     git -C "$1" config --local --unset githooks.autoupdate.registered
 
+    legacy_transformation_adjust_local_paths "$1" || return 1
+
+    return 0
+}
+
+#####################################################
+# Check if `$1` is not a supported git clone url and
+#   is treated as a local path to a repository.
+#   See `https://tools.ietf.org/html/rfc3986#appendix-B`
+
+# Returns: 0 if it is a local path, 1 otherwise
+#####################################################
+is_local_path() {
+    if echo "$1" | grep -Eq "^[^:/?#]+://" ||  # its a <scheme>://
+        echo "$1" | grep -Eq "^.+@.+:.+"; then # or its a short scp syntax
+        return 1
+    fi
     return 0
 }
 
+#####################################################
+# Check if url `$1` is a local url, e.g `file://`.
+#
+# Returns: 0 if it is a local url, 1 otherwise
+#####################################################
+is_local_url() {
+    if echo "$1" | grep -Eq "^\s*file://"; then
+        return 0
+    fi
+    return 1
+}
+
+############################################################
+# Function to adjust local paths in `.githooks/.shared`
+#   file which are forbidden.
+#
+# Returns:
+#   1 when failed, 0 otherwise
+############################################################
+legacy_transformation_adjust_local_paths() {
+
+    if [ "$(git -C "$1" rev-parse --is-bare-repository 2>/dev/null)" = "true" ]; then
+        SHARED_FILE="$1/.githooks/.shared"
+    else
+        SHARED_FILE="$1/../.githooks/.shared"
```

Comment:
What is `$1` here? Is it not the repo root? Does it actually have `repo_root/.git` in that?

Reply:
Jeah, its always the GIT_DIR.